aten/src/ATen/native/NamedTensor.cpp

```diff
@@ -142,12 +142,36 @@ static Tensor align(const Tensor& tensor, DimnameList names, bool is_aligning_tw
 }
 
 Tensor align_to(const Tensor& tensor, DimnameList names) {
-  TORCH_CHECK(
-      names.size() >= tensor.dim(),
-      "Cannot align tensor with dims ",
-      tensor.names(),
-      " to a shorter list of dims ", names, ".");
-  return align(tensor, names, /*aligning_two_tensors=*/false);
+  auto tensor_names = tensor.names();
+  auto tensor_sizes = tensor.sizes();
+  auto tensor_strides = tensor.strides();
+  std::vector<int64_t> new_sizes(names.size(), 1);
+  std::vector<int64_t> new_strides(names.size(), 0);
+
+  for (auto idx = 0; idx < tensor_names.size(); ++idx) {
+    const auto& dim = tensor_names[idx];
+    TORCH_CHECK(dim.is_normal(),
+        "align_to: All input dims must be named. Found unnamed dim at index ",
+        dim, " of Tensor", tensor_names);
+    auto it = std::find(names.begin(), names.end(), dim);
+    TORCH_CHECK(it != names.end(),
+        "align_to: Cannot find dim ", dim, " from Tensor", names,
+        " in desired alignment ", names, ".");
```

Comment:
do we usually end our error messages in a period? I notice one of these errors does and one doesn't.

Reply:
Some of our error messages end in a period, some of them don't. I can make both of these match but we shouldn't block this PR on that if it's the only problem

Reply:
yeah not a blocker just a nit/question, would be nice to have a consistent convention for us to follow though.